File: app/src/main/java/com/app/framework/common/ui/BaseActivity.java

package com.app.framework.common.ui;


import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.app.framework.net.ApiService;
import com.app.framework.mvp.BasePresenter;
import com.app.framework.net.HttpModule;
import com.gyf.immersionbar.ImmersionBar;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public abstract class BaseActivity extends AppCompatActivity implements BasePresenter {

    private CompositeDisposable compositeDisposable;
    public ApiService apiService = HttpModule.Companion.getInstance().getService();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ImmersionBar.with(this).statusBarDarkFont(true).init();
    }

    @Override
    public void bye() {
        finish();
    }

    @Override
    public void onResume() {
        super.onResume();
    }

    @Override
    public void onPause() {
        super.onPause();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
//        ImmersionBar.with(this); //必须调用该方法，防止内存泄漏
        if (compositeDisposable != null) {
            compositeDisposable.clear();
        }
    }

    protected void addSubscribe(Disposable disposable) {
        if (compositeDisposable == null) {
            compositeDisposable = new CompositeDisposable();
        }
        compositeDisposable.add(disposable);
    }

}

File: mvvmdemo/src/main/java/com/huxh/mvvmdemo/base/activity/BaseActivity.kt
package com.huxh.mvvmdemo.base.activity

import android.os.Bundle
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatActivity

abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentLayout()
    }

    open fun setContentLayout() {
        setContentView(layoutId())
        onViewCreate()
        initData()
    }

    open fun initData() {}

    abstract fun onViewCreate()

    @LayoutRes
    abstract fun layoutId(): Int

    open fun showLoading() {}

    open fun dismissLoading() {}
}
